### PR TITLE
Add additional filters to operator hub and developer catalog

### PR DESCRIPTION
**Pull request by jeff-phillips-18**
Add Provider Type and Installed filters to operator hub.
Add the Installed Operators to the developer catalog type filter.

This required a rework of the tile pages. Changed the catalog and operator-hub tile pages to extend the tile-view-page in order to be able to override how the filters are shown (allowing the hide/show of the Community provider types in operator hub). Notice rather than passing in the functions the pages now override the default functions in the tile view page.

https://jira.coreos.com/browse/MKTPLC-174
https://jira.coreos.com/browse/MKTPLC-175

UI changes are documented in https://docs.google.com/document/d/1FTvPacdQFpAdrH7qrZWmcbUAqBmcNb5BAqdbT3OOGf0/edit?ts=5c2ed2df

@serenamarie125 @tlwu2013

### CODE REVIEW

File: frontend/public/components/catalog/catalog-items.jsx [hunk 1]
```diff
@@ -66,79 +67,103 @@ export const catalogCategories = {
...
-
 const setURLParams = params => {
   const url = new URL(window.location);
   const searchParams = `?${params.toString()}${url.hash}`;
 
   history.replace(`${url.pathname}${searchParams}`);
 };
 
-export class CatalogTileViewPage extends React.Component {
+export class CatalogTileViewPage extends TileViewPage {
```
Comment (spadgett) [inline frontend/public/components/catalog/catalog-items.jsx]: I'm concerned with using inheritance here. It's considered an antipattern in react and has caused us some pain in other places (particularly when we need to connect components to Redux). Is there a way we can use composition instead?

https://reactjs.org/docs/composition-vs-inheritance.html

Comment (jeff-phillips-18) [reply frontend/public/components/catalog/catalog-items.jsx]: I'll take a shot at it.

**Comment by jeff-phillips-18** [comment]
closing in favor of https://github.com/openshift/console/pull/1051

File: frontend/public/components/operator-hub/operator-hub-page.tsx [hunk 2]
```diff
@@ -43,15 +46,32 @@ const normalizePackageManifests = (packageManifests: PackageManifestKind[] = [])
     longDescription = longDescription || _.get(packageManifest, 'status.channels[0].currentCSVDesc.description');
     const catalogSource = _.get(packageManifest, 'status.catalogSource');
     const catalogSourceNamespace = _.get(packageManifest, 'status.catalogSourceNamespace');
+    let providerType;
+    switch (srcProvider) {
+      case 'redhat':
+        providerType = 'Red Hat';
+        break;
+      case 'redhat-partner':
+        providerType = 'Red Hat Partner';
```
Comment (galletti94) [inline frontend/public/components/operator-hub/operator-hub-page.tsx]: Should be Certified or Red Hat Certified

**Pull request closed without merging**